index.js

```diff
@@ -8,6 +9,16 @@ const notes = []
 
 app.use(jsonParser)
 
+app.put('/notes/:id', putValidator)
+
+app.put('/notes/:id', (req, res) => {
+  const noteId = parseInt(req.params.id, 10)
+  const note = notes.find(note => note.id === noteId)
+  if (!req.body) return res.sendStatus(404)
```

Comment:
This will never be `true`. Body parser ensures that there is at least an empty request body.